Cubical/HITs/Sn/Properties.agda

```diff
@@ -13,29 +12,329 @@ open import Cubical.Foundations.Prelude
+open import Cubical.Homotopy.Loopspace
+open import Cubical.Homotopy.Connected
 
-open Iso
+sphereConnected : (n : HLevel) → isConnected (suc n) (S₊ n)
+sphereConnected n = ∣ ptSn n ∣ , (elim (λ _ → isOfHLevelPath (suc n) (isOfHLevelTrunc (suc n)) _ _)
+                                      (λ a → sym (spoke ∣_∣ (ptSn n)) ∙ spoke ∣_∣ a))
+
+-- Elimination principles for spheres
+sphereElim : ∀ {ℓ} (n : ℕ) {A : (S₊ (suc n)) → Type ℓ} → ((x : S₊ (suc n)) → isOfHLevel (suc n) (A x))
+          → A (ptSn (suc n))
+          → (x : S₊ (suc n)) → A x
+sphereElim zero hlev pt = toPropElim hlev pt
+sphereElim (suc n) hlev pt north = pt
+sphereElim (suc n) {A = A} hlev pt south = subst A (merid (ptSn (suc n))) pt
+sphereElim (suc n) {A = A} hlev pt (merid a i) = helper i
+  where
+  helper : PathP (λ i → A (merid a i)) pt (subst A (merid (ptSn (suc n))) pt)
+  helper = sphereElim n {A = λ a → PathP (λ i → A (merid a i)) pt (subst A (merid (ptSn (suc n))) pt)}
+                        (λ a → isOfHLevelPathP' (suc n) (hlev south) _ _)
+                        (λ i → transp (λ j → A (merid (ptSn (suc n)) (i ∧ j))) (~ i) pt)
+                        a
+
+sphereElim2 : ∀ {ℓ} (n : ℕ) {A : (S₊ (suc n)) → (S₊ (suc n)) → Type ℓ}
+          → ((x y : S₊ (suc n)) → isOfHLevel (suc n) (A x y))
+          → A (ptSn (suc n)) (ptSn (suc n))
+          → (x y : S₊ (suc n)) → A x y
+sphereElim2 n hlev pt = sphereElim n (λ _ → isOfHLevelΠ (suc n) λ _ → hlev _ _)
+                                     (sphereElim n (hlev _ ) pt)
 
 private
-  variable
-    ℓ ℓ' : Level
-    A : Type ℓ
-    B : Type ℓ'
+  compPath-lem : ∀ {ℓ} {A : Type ℓ} {x y z : A} (p : x ≡ y) (q : z ≡ y)
+              → PathP (λ i → (p ∙ sym q) i ≡ y) p q
+  compPath-lem {y = y} p q i j =
+    hcomp (λ k → λ { (i = i0) → p j
+                    ; (i = i1) → q (~ k ∨ j)
+                    ; (j = i1) → y })
+          (p (j ∨ i))
+
+-- Elimination rule for fibrations (x : Sⁿ) → (y : Sᵐ) → A x y of h-Level (n + m).
+-- The following principle is just the special case of the "Wedge Connectivity Lemma"
+-- for spheres (See Cubical.Homotopy.WedgeConnectivity or chapter 8.6 in the HoTT book).
+-- We prove it directly here for two reasons:
+-- (i) it should perform better
+-- (ii) we get a slightly stronger statement for spheres -- One of the homotopies will, by design, be refl
+
+wedgeConSn : ∀ {ℓ} (n m : ℕ) {A : (S₊ (suc n)) → (S₊ (suc m)) → Type ℓ}
+          → ((x : S₊ (suc n)) (y : S₊ (suc m)) → isOfHLevel ((suc n) + (suc m)) (A x y))
+          → (f : (x : _) → A (ptSn (suc n)) x)
+          → (g : (x : _) → A x (ptSn (suc m)))
+          → (g (ptSn (suc n)) ≡ f (ptSn (suc m)))
+          → Σ[ F ∈ ((x : S₊ (suc n)) (y : S₊ (suc m)) → A x y) ]
+              ((x : S₊ (suc m)) → F (ptSn (suc n)) x ≡ f x) × ((x : S₊ (suc n)) → F x (ptSn (suc m)) ≡ g x)
+wedgeConSn zero zero {A = A} hlev f g hom = F , (λ _ → refl) , right
+  where
+  helper : SquareP (λ i j → A (loop i) (loop j)) (cong f loop) (cong f loop)
```

Comment:
I would use `_∙∙_∙∙_` instead of these hcomp's

Reply:
This applies to all of the hcomps where you attach 2 lines to another line in this proof

Reply:
These are all dependent paths, so I can't use `_∙∙_∙∙_`

Reply:
Ahh, right! My bad